### PR TITLE
User handling

### PR DESCRIPTION
**Pull request by matkrzeminski**
These changes allow for:
- Logging in with GitHub.
- Creating users.json and adding data of this format to it: `{
        "github_id": int,
        "username": str,
        "email": str,
        "avatar_url": str,
        "role": str
    }` which is used for the permissions.
- Using access and refresh JWT.
- Adding roles to users (proofreader and translator logic for now).

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -33,4 +33,11 @@ ES_HOST=elasticsearch
...
-LICENSE=basic
\ No newline at end of file
+LICENSE=basic
+
+# Github
+GITHUB_ACCESS_TOKEN_URL=https://github.com/login/oauth/access_token
+GITHUB_USER_URL=https://api.github.com/user
+GITHUB_AUTHORIZE_URL=https://github.com/login/oauth/authorize
+GITHUB_CLIENT_ID=change_me
+GITHUB_CLIENT_SECRET=change_me
```
Comment (marekbryling) [inline .env.example:43]: Missing empty line

File: Makefile [hunk 2]
```diff
@@ -24,3 +24,6 @@ build:
 
 down:
 	@$(COMPOSE) down
+
+test:
+	@$(COMPOSE) exec bilara-backend pytest /app/app/tests
```
Comment (marekbryling) [inline Makefile:29]: Missing empty line

**Review by marekbryling — CHANGES_REQUESTED** [review]

**Review by marekbryling — APPROVED** [review]

File: backend/app/services/users/utils.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+def get_user(github_id: int) -> UserData:
+    users: list[dict[str, str]] = get_json_data(settings.USERS_FILE)
+    return UserData(**[user for user in users if user["github_id"] == github_id][0])
+
+
+def is_username_in_muid(username: str, muid: str) -> bool:
+    if not username or not muid:
+        return False
+    return username == muid.split("-", 2)[2]
```
Comment (mkorlaga) [inline backend/app/services/users/utils.py:17]: `-` can be used in github username, muid.split() can eventually lead to not full username

File: backend/app/services/auth/utils.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        raise get_credentials_exception()
+    return token_data
+
+
+def get_credentials_exception() -> HTTPException:
+    return HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Could not validate credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
```
Comment (mkorlaga) [inline backend/app/services/auth/utils.py:48]: This forces us to check what is the `get_credentials_exception` - creating separate file for exceptions and creating new one based on HTTPException would be more clear `class UnauthorizedException`

File: backend/app/services/users/schema.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from pydantic import BaseModel, EmailStr
+
+
+class UserData(BaseModel):
+    github_id: int
+    username: str
+    email: EmailStr
+    avatar_url: str
+    role: str = "proofreader"
```
Comment (mkorlaga) [inline backend/app/services/users/schema.py:9]: This can be enumed from `backend/app/services/users/roles.py`

**Pull request merged**